solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,648 @@
+        uint256 signedQuoteLength,
+        uint256 executorValue
+    );
+
+    /// @notice Emitted when tokens are transferred via NTT Manager With Executor
+    /// @param amount Amount of tBTC transferred
+    /// @param destinationChain Wormhole chain ID of the destination
+    /// @param actualRecipient Actual recipient address on destination chain
+    /// @param transferSequence NTT transfer sequence number
+    /// @param encodedReceiver Original encoded receiver data
+    /// @param executorCost Cost paid to executor service in wei
+    event TokensTransferredNttWithExecutor(
+        uint256 amount,
+        uint16 destinationChain,
+        bytes32 actualRecipient,
+        uint64 transferSequence,
+        bytes32 encodedReceiver,
+        uint256 executorCost
+    );
+
+    /// @notice Emitted when a destination chain is added or removed
+    event SupportedChainUpdated(uint16 indexed chainId, bool supported);
+
+    /// @notice Emitted when default supported chain is updated
+    event DefaultSupportedChainUpdated(uint16 indexed chainId);
+
+    /// @notice Emitted when default parameters are updated
+    event DefaultParametersUpdated(
+        uint256 gasLimit,
+        uint16 feeBps,
+        address feeRecipient
+    );
+
+    /// @notice Emitted when the underlying NTT Manager is updated
+    event UnderlyingNttManagerUpdated(
+        address indexed oldManager,
+        address indexed newManager
+    );
+
+    /// @notice Emitted when the NTT Manager With Executor is updated
+    event NttManagerWithExecutorUpdated(
+        address indexed oldManager,
+        address indexed newManager
+    );
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the L1 depositor contract with executor support
+    /// @param _tbtcBridge tBTC Bridge contract address
+    /// @param _tbtcVault tBTC Vault contract address
+    /// @param _nttManagerWithExecutor NTT Manager With Executor contract address
+    /// @param _underlyingNttManager Underlying NTT Manager contract address
+    function initialize(
+        address _tbtcBridge,
+        address _tbtcVault,
+        address _nttManagerWithExecutor,
+        address _underlyingNttManager
+    ) external initializer {
+        __AbstractL1BTCDepositor_initialize(_tbtcBridge, _tbtcVault);
+        __Ownable_init();
+
+        require(
+            _nttManagerWithExecutor != address(0),
+            "NTT Manager With Executor address cannot be zero"
+        );
+        require(
+            _underlyingNttManager != address(0),
+            "Underlying NTT Manager address cannot be zero"
+        );
+
+        nttManagerWithExecutor = INttManagerWithExecutor(
+            _nttManagerWithExecutor
+        );
+        underlyingNttManager = _underlyingNttManager;
+
+        // Set reasonable defaults
+        defaultDestinationGasLimit = 500000; // 500k gas for destination execution
```

Comment:
I'm being very nitpicky here, but we use some magic numbers/constants here and there, we might want to put them into `private constant` contract variables instead